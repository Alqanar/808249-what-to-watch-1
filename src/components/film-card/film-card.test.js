import React from "react";
import renderer from "react-test-renderer";

import FilmCard from "./film-card.tsx";


const film = {
    id: `0`,
    name: `Avengers: Endgame`,
    posterLink: `https://st.kp.yandex.net/images/film_iphone/iphone360_843650.jpg`,
    trailer: `https://youtu.be/TcMBFSGVi1c`
};

describe(`Film card correctly renders`, () => {
    it(`when isMainPage true`, () => {
        const tree = renderer
            .create(<FilmCard
                movie={film}
                key={0}
                isMainPage={true}
            />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isMainPage false`, () => {
        const tree = renderer
            .create(<FilmCard
                movie={film}
                key={0}
                isMainPage={false}
            />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
