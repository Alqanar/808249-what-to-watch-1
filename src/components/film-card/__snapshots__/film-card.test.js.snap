// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film card correctly renders 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <button
    className="small-movie-card__play-btn"
    type="button"
  >
    Play
  </button>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Avengers: Endgame"
      height="175"
      src="https://st.kp.yandex.net/images/film_iphone/iphone360_843650.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Avengers: Endgame
    </a>
  </h3>
</article>
`;
